src/auth_ldap.c

```
@@ -632,13 +636,32 @@ checkldapauth(struct ldap_auth_request *request)
 			}
 		}
 
+		/*
+		 * If the password of LDAP authentication is encrypted (ldap_password == "$bindpasswd"),
+		 * then decrypt it, replace the variable and continue using the real password.
+		 */
+		if (ldap_password != NULL && strcmp(ldap_password, "$bindpasswd") == 0 && request->client->ldap_key != NULL) {
```

If request->client->ldap_key is NULL, I think we need to report an error, or if you have tested this case that cf_auth_key_file is not set a value, I think the authentication will fail, but we need to log an error.